pkg/kubectl/cmd/taint.go

```diff
@@ -113,42 +120,56 @@ func NewCmdTaint(f cmdutil.Factory, out io.Writer) *cobra.Command {
 
 // reorganizeTaints returns the updated set of taints, taking into account old taints that were not updated,
 // old taints that were updated, old taints that were deleted, and new taints.
-func reorganizeTaints(obj runtime.Object, overwrite bool, taintsToAdd []v1.Taint, taintsToRemove []v1.Taint) ([]v1.Taint, error) {
-	node, ok := obj.(*v1.Node)
-	if !ok {
-		return nil, fmt.Errorf("unexpected type %T, expected Node", obj)
-	}
-
+func reorganizeTaints(node *v1.Node, overwrite bool, taintsToAdd []v1.Taint, taintsToRemove []v1.Taint) (string, []v1.Taint, error) {
```

Comment:
So this should honestly be a topic of conversation for the sig today.  Taints are probably the best example I can think of that has logic sprawl.  

There are changes in api, admission, node, scheduler, controllers, kubectl and TBH it needs to be centralized.  A change in one area will affect another, in it's current implementation I almost consider it a bug-factory. 

/cc @davidopp @gmarek